Source/UI/NewUI/BillingForm/ViewController/BillingFormViewController.swift

```diff
@@ -0,0 +1,171 @@
+                constant: 20)
+        ])
+    }
+    
+    private func setUpKeyboard() {
+        title = "addressViewControllerTitle".localized(forClass: AddressViewController.self)
+        registerKeyboardHandlers(notificationCenter: notificationCenter,
+                                      keyboardWillShow: #selector(keyboardWillShow),
+                                      keyboardWillHide: #selector(keyboardWillHide))
+        
+//        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(self.dismissKeyboard (_:)))
+//        view.addGestureRecognizer(tapGesture)
+//        NotificationCenter.default.addObserver(self, selector: #selector(BillingFormViewController.keyboardWillShow), name: UIResponder.keyboardWillShowNotification, object: nil)
+//        NotificationCenter.default.addObserver(self, selector: #selector(BillingFormViewController.keyboardWillHide), name: UIResponder.keyboardWillHideNotification, object: nil)
+    }
+}
+
+// UITableViewDataSource
+extension BillingFormViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        delegate?.tableView(numberOfRowsInSection: section) ?? 0
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        delegate?.tableView(cellForRowAt: indexPath, sender: self) ?? UITableViewCell()
+    }
+    
+    func tableView(_ tableView: UITableView, estimatedHeightForRowAt indexPath: IndexPath) -> CGFloat {
+        delegate?.tableView(estimatedHeightForRowAt: indexPath) ?? 0.0
+    }
+}
+
+//FormCellDelegate
+extension BillingFormViewController: BillingFormTextFieldCellDelegate {
+    func textFieldDidChangeCharacters(textField: UITextField, replacementString: String) {
+        guard let textField = textField as? BillFormTextField else { return }
+        delegate?.textFieldIsChanged(textField: textField, replacementString: replacementString)
+    }
+    
+    func textFieldDidEndEditing(textField: UITextField) {
+        guard let textField = textField as? BillFormTextField else { return }
+        delegate?.validate(textField: textField)
+    }
+    
+    
+    func textFieldShouldBeginEditing(textField: UITextField) {
+        self.focusedTextField = textField
+    }
+    
+    func textFieldShouldReturn() {
+        view.endEditing(true)
+    }
+}
+
+//FormHeaderCellDelegate
+extension BillingFormViewController: BillingFormHeaderCellDelegate {
+    
+    func doneButtonIsPressed() {
+        delegate?.doneButtonIsPressed(sender: self)
+    }
+    
+    func cancelButtonIsPressed() {
+        delegate?.cancelButtonIsPressed(sender: self)
+    }
+}
+
+// ViewModel
+extension BillingFormViewController {
+    private func setupViewModel() {
+        viewModel.updateRow = {
+            DispatchQueue.main.async { [weak self] in
+                guard let self = self, let row = self.viewModel.updatedRow else { return }
```

Comment:
Yes.
https://github.com/apple/swift/pull/15306

Reply:
I am not sure I follow the relevance of that post. I am not asking what the syntax `guard let self = self` does, I am asking why do we need to have a strong referenced self in this case?

We can make use of the guard that checks if the `updatedRow` is not nil, to also validate `self` is not nil, without implicitly capturing self in the closure:

```
viewModel.updateRow = {
             DispatchQueue.main.async { [weak self] in
                 guard let row = self?.viewModel.updatedRow else { return }
                 let indexPath = IndexPath(row: row, section: 0)
                 self?.tableView.reloadRows(at: [indexPath], with: .automatic)
             }
         }
```

I may be mistaken as I didn't check inside compiler, but this code should work just the same, whilst continuing to hold a weak self reference.

Happy to get into it further on the benefits of this instead of automatically defaulting to strong self captures every time.